daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/ScriptLedgerClient.scala

```diff
@@ -73,7 +73,11 @@ object ScriptLedgerClient {
 // us to plug in something that interacts with the JSON API as well as
 // something that works against the gRPC API.
 trait ScriptLedgerClient {
-  def query(parties: OneAnd[Set, Ref.Party], templateId: Identifier)(implicit
+  def query(
+      parties: OneAnd[Set, Ref.Party],
+      templateId: Identifier,
+      enableContractUpgrading: Boolean = false,
```

Comment:
perhaps this should be a property of `ScriptLedgerClient` instead of an argument to `query`/`queryContractId`, but I don't feel too strongly about that

Reply:
Looking further at how this'd need to be passed around, since the clients are created before the ScriptF Env, where `timeMode` is currently placed, I think i'd rather keep as is. This also keeps us open to the possibility of a strictQuery later